### PR TITLE
Add circuit breaker configuration support

### LINKED ISSUE #238: [Task] Add circuit breaker configuration support for PHP
### Description

Add `ClientCircuitBreakerConfiguration` to the client configuration, matching the existing implementation in Java, Python, Node, and Go (valkey-io/valkey-glide#6050). The circuit breaker logic lives in the Rust core, this task only requires adding the config class, serializing it to protobuf, and surfacing a `CircuitBreakerException` type to callers.

### Checklist

- [ ] Add `ClientCircuitBreakerConfiguration` class
- [ ] Serialize config to protobuf `ConnectionRequest`
- [ ] Add `CircuitBreakerException` type
- [ ] Add unit test for config serialization

### Related

- valkey-io/valkey-glide#6050
- valkey-io/valkey-glide#5996

### PR DESCRIPTION
**Pull request by jeremyprime**
### Summary

Add circuit breaker configuration support to the PHP client, matching the existing implementation in Java, Python, Node, and Go (valkey-io/valkey-glide#6050). The circuit breaker logic lives in the Rust core; this change adds the config array parameter, serializes it to protobuf, and surfaces a `CircuitBreakerException` type to callers.

### Issue link

Closes #238

### Features / Behaviour Changes

- New `?array $circuit_breaker` parameter on `ValkeyGlide::connect()` and `ValkeyGlideCluster::__construct()`
- New `CircuitBreakerException` class (extends `ValkeyGlideException`) thrown when the circuit breaker is open
- Configuration keys: `window_size_ms`, `failure_rate_threshold`, `min_errors`, `open_timeout_ms`, `count_timeouts`, `consecutive_successes`
- All parameters optional with sensible defaults matching other clients

### Implementation

- `common.h`: Added `valkey_glide_circuit_breaker_config_t` struct, constants, and plumbed into base config
- `valkey_glide.c`: Parse circuit breaker array, register exception class, cleanup on free
- `valkey_glide_core_commands.c`: Serialize config to `ClientCircuitBreakerConfig` protobuf message (field 30)
- `valkey_glide_commands_common.h`: Dispatch `CircuitBreakerException` when `command_error_type == CircuitBreakerOpen`
- Stubs and mock constructors updated for both standalone and cluster

### Testing

- New unit tests added to `ConnectionRequestTest.php`:
  - `testCircuitBreakerStandaloneDefaults`: empty array uses defaults
  - `testCircuitBreakerClusterDefaults`: same for cluster
  - `testCircuitBreakerStandaloneCustomValues`: all fields customized
  - `testCircuitBreakerClusterCustomValues`: same for cluster
  - `testCircuitBreakerNotSetByDefault`: null when not provided

### Checklist

- [x] This Pull Request is related to one issue.
- [x] Commit message has a detailed description of what changed and why.
- [x] Tests are added or updated.
- [x] CHANGELOG.md and documentation files are updated.
- [x] Destination branch is correct - main or release
- [x] Create merge commit if merging release branch into main, squash otherwise.

### CODE REVIEW

File: valkey_glide_commands_common.h [hunk 1]
```diff
@@ -74,8 +82,8 @@ static inline void handle_command_result_or_throw(CommandResult* result,
         return;
     }
     if (result->command_error) {
-        zend_throw_exception(
-            get_valkey_glide_exception_ce(), result->command_error->command_error_message, 0);
+        zend_class_entry* exception_ce = get_exception_ce_for_command_error(result->command_error);
+        zend_throw_exception(exception_ce, result->command_error->command_error_message, 0);
```
Comment (prateek-kumar-improving) [inline valkey_glide_commands_common.h]: Should all zend_throw_exception method calls in the codebase be updated to use `get_exception_ce_for_command_error()`? Can the `command_error_type` be returned at other zend_throw_exception() calls that are not updated?

File: tests/ConnectionRequestTest.php [hunk 2]
```diff
@@ -742,4 +743,94 @@ private function get_addresses(\Connection_request\ConnectionRequest $request):
     {
         return array_map(fn($a) => $a->getHost(), iterator_to_array($request->getAddresses()));
     }
+
+    // ================================================================
+    // Circuit Breaker Tests
+    // ================================================================
```
Comment (prateek-kumar-improving) [inline tests/ConnectionRequestTest.php:749]: Please add edge-case tests for invalid values.

File: valkey_glide.c [hunk 3]
```diff
@@ -573,6 +579,60 @@ int valkey_glide_build_client_config_base(valkey_glide_php_common_constructor_pa
         config->client_side_cache = NULL;
     }
```
Comment (prateek-kumar-improving) [inline valkey_glide.c:581]: Can we please add validations for the fields.

```
- ['window_size_ms' => -1] → silently wraps to 4294967295 (uint32 overflow)
- ['window_size_ms' => 0] → passes through as 0
- ['failure_rate_threshold' => 2.0] → accepted, but a 200% failure rate makes no sense
- ['failure_rate_threshold' => 0.0] → accepted, but the circuit would never trip
- ['min_errors' => -5] → wraps to a uint32
- ['consecutive_successes' => 0] → accepted
```

Python adds these checks:
```
if window_size_ms <= 0:
    raise ValueError("window_size_ms must be positive")
if not (0.0 < failure_rate_threshold <= 1.0):
    raise ValueError("failure_rate_threshold must be between 0.0 (exclusive) and 1.0 (inclusive)")
if min_errors <= 0:
    raise ValueError("min_errors must be positive")
if open_timeout_ms <= 0:
    raise ValueError("open_timeout_ms must be positive")
if consecutive_successes <= 0:
    raise ValueError("consecutive_successes must be positive")
```

File: common.h [hunk 4]
```diff
@@ -145,6 +145,31 @@ typedef struct {
...
+#define VALKEY_GLIDE_CB_COUNT_TIMEOUTS "count_timeouts"
+#define VALKEY_GLIDE_CB_CONSECUTIVE_SUCCESSES "consecutive_successes"
+
+#define VALKEY_GLIDE_CB_DEFAULT_WINDOW_SIZE_MS 10000
+#define VALKEY_GLIDE_CB_DEFAULT_FAILURE_RATE_THRESHOLD 0.5
+#define VALKEY_GLIDE_CB_DEFAULT_MIN_ERRORS 50
+#define VALKEY_GLIDE_CB_DEFAULT_OPEN_TIMEOUT_MS 5000
+#define VALKEY_GLIDE_CB_DEFAULT_COUNT_TIMEOUTS false
+#define VALKEY_GLIDE_CB_DEFAULT_CONSECUTIVE_SUCCESSES 3
+#define VALKEY_GLIDE_ERROR_TYPE_CIRCUIT_BREAKER_OPEN 4
```
Comment (currantw) [inline common.h:162]: Should we define/populate circuit breaker defaults at the client level – or should we let the Rust core set them? I am doing the latter on the C# client, so want to make sure these are aligned!

File: valkey_glide_ft_common.c [hunk 5]
```diff
@@ -53,8 +53,9 @@ int execute_ft_command_internal(const void*      glide_client,
         return 0;
     }
     if (result->command_error) {
-        zend_throw_exception(
-            get_valkey_glide_exception_ce(), result->command_error->command_error_message, 0);
+        zend_throw_exception(get_exception_ce_for_command_error(result->command_error),
+                             result->command_error->command_error_message,
+                             0);
```
Comment (currantw) [inline valkey_glide_ft_common.c]: Since `get_exception_ce_for_command_error` seems to only be used in this format, would it make sense to just wrap this all into something like `throw_exception_for_command_error()`?

File: tests/ConnectionRequestTest.php (hunk 2 above)
Comment (currantw) [inline tests/ConnectionRequestTest.php:749]: It seems like you have only added unit tests, not any integration tests. Do you think that is sufficient? I tried to add some basic integration tests (basically just that connection works, that the circuit breaker could be tripped with low thresholds) for C#.

**Review by currantw — COMMENTED** [review]
Reviewed to 7ba5343.

File: tests/ConnectionRequestTest.php (hunk 2 above)
Comment (jeremyprime) [reply tests/ConnectionRequestTest.php:749]: There are no edge case tests for any invalid options in any of the clients.  If we add them it will be a cross-client effort (see https://github.com/valkey-io/valkey-glide/issues/5937), not as part of this PR.

File: valkey_glide_commands_common.h (hunk 1 above)
Comment (jeremyprime) [reply valkey_glide_commands_common.h]: There are 2 types of `zend_throw_exception` here, FFI command errors (which have `command_error_type`) and local validation errors which are thrown before reaching the Rust core. All FFI error paths use  `get_exception_ce_for_command_error`. The remaining ones are local and have no `CommandError` and can never return `CircuitBreakerOpen`.

File: valkey_glide.c (hunk 3 above)
Comment (jeremyprime) [reply valkey_glide.c:581]: As mentioned above, since no other config options have edge case validation I would rather see this done across all clients as part of https://github.com/valkey-io/valkey-glide/issues/5937.

File: common.h (hunk 4 above)
Comment (jeremyprime) [reply common.h:162]: I think we should define and send from client (Java, Node, Python, Go already doing this too).  This reduces ambiguity and documents from client side what is expected, although does duplicate the values in core and client.  This is also how other features are handled (reconnection, compression, etc).

File: tests/ConnectionRequestTest.php (hunk 2 above)
Comment (jeremyprime) [reply tests/ConnectionRequestTest.php:749]: Only Java has CB integration tests, other clients just have unit tests for CB.  I did this because CB is a bit more complex and I thought it made more sense to test the tripping and other scenarios as system tests (in our internal system test suite) instead.  Especially since some of those scenarios can be a bit flaky in an integration test.

File: valkey_glide_ft_common.c (hunk 5 above)
Comment (jeremyprime) [reply valkey_glide_ft_common.c]: Done, now uses a common `throw_command_error`.

**Review by prateek-kumar-improving — APPROVED** [review]

File: valkey_glide.c (hunk 3 above)
Comment (currantw) [reply valkey_glide.c:581]: Sounds good to me. I will update C# accordingly.

File: common.h (hunk 4 above)
Comment (currantw) [reply common.h:162]: Sounds good. I agree that the visibility of default values to the user is worth the duplication.

File: valkey_glide_ft_common.c (hunk 5 above)
Comment (currantw) [reply valkey_glide_ft_common.c]: Thank you 🙏

File: tests/ConnectionRequestTest.php (hunk 2 above)
Comment (currantw) [reply tests/ConnectionRequestTest.php:749]: Sounds good. I will update C# client accordingly.

**Review by currantw — APPROVED** [review]
Reviewed to cec7239.
Looks good ✅ 
Thanks for replying to all my comments.
Will update C# GLIDE client to use a consistent approach.
